### PR TITLE
universal-gcode-sender: Add version 2.0.7

### LINKED ISSUE #5849: [Request] Add universal-gcode-sender
[UGS](https://github.com/winder/Universal-G-Code-Sender) is a tool for 3d-printing and CNC.

### PR DESCRIPTION
**Pull request by MunsterPlop**
* Closes #5849

The archive contains both 32 and 64bit executables and I originally just created shortcuts for both and left both shims available
but I decided to make only the detected architecture available.
Not sure if it is okay or ugly stuff, so let me know if I should just stick to the former option.

### CODE REVIEW

File: bucket/universal-gcode-sender.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            ],
+            "shortcuts": [
+                [
+                    "bin/ugsplatform64.exe",
+                    "UGS Platform"
+                ]
+            ]
+        }
+    },
+    "extract_dir": "ugsplatform-win",
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:33]:
```suggestion
```

File: bucket/universal-gcode-sender.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+    "version": "2.0.7",
+    "description": "Full featured gcode platform used for interfacing with advanced CNC controllers.",
+    "homepage": "https://winder.github.io/ugs_website/",
+    "license": "GPL-3",
+    "url": "https://ugs.jfrog.io/ugs/UGS/v2.0.7/ugs-platform-app-win.zip",
+    "hash": "23b0726c0b4b9d29a1a2adaab53f1a16cd4c45c16413bfa67c1900843f46399f",
+    "architecture": {
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:8]:
```suggestion
    "architecture": {
    "extract_dir": "ugsplatform-win",
```

File: bucket/universal-gcode-sender.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+    "version": "2.0.7",
+    "description": "Full featured gcode platform used for interfacing with advanced CNC controllers.",
+    "homepage": "https://winder.github.io/ugs_website/",
+    "license": "GPL-3",
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:5]: invalid SPDX identifier

File: bucket/universal-gcode-sender.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        "64bit": {
+            "bin": [
+                [
+                    "bin/ugsplatform64.exe",
+                    "ugsplatform"
+                ]
+            ],
+            "shortcuts": [
+                [
+                    "bin/ugsplatform64.exe",
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:27]: backslashes

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/universal-gcode-sender.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{
+    "version": "2.0.7",
+    "description": "Full featured gcode platform used for interfacing with advanced CNC controllers.",
+    "homepage": "https://winder.github.io/ugs_website/",
+    "license": "GPL-3.0",
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:5]: Invalid SPDX identifier

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/universal-gcode-sender.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
     "version": "2.0.7",
     "description": "Full featured gcode platform used for interfacing with advanced CNC controllers.",
     "homepage": "https://winder.github.io/ugs_website/",
-    "license": "GPL-3.0",
+    "license": "GPL-3.0-or-later",
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:5]: Github licenses are not explicitly `or-later`. Unless you have explicit proof on authors website it is later, you cannot use it.

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/universal-gcode-sender.json (hunk 6 above)
Comment (MunsterPlop) [reply bucket/universal-gcode-sender.json]: Alright, good to know.

File: bucket/universal-gcode-sender.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+                    "UGS Platform"
+                ]
+            ]
+        }
+    },
+    "checkver": {
+        "github": "https://github.com/winder/Universal-G-Code-Sender"
+    },
+    "autoupdate": {
+        "url": "https://ugs.jfrog.io/ugs/UGS/v$version/ugs-platform-app-win.zip"
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:38]:
```suggestion
        "url": "https://ugs.jfrog.io/ugs/UGS/v$version/ugs-platform-app-win.zip",
        "hash": {
            "url": "$url.sha256"
        }
```

File: bucket/universal-gcode-sender.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        "32bit": {
+            "bin": "bin\\ugsplatform.exe",
+            "shortcuts": [
+                [
+                    "bin\\ugsplatform.exe",
+                    "UGS Platform"
+                ]
+            ]
+        },
+        "64bit": {
```
Comment (Ash258) [inline bucket/universal-gcode-sender.json:19]: 64bit first

**Review by Ash258 — CHANGES_REQUESTED** [review]
image

**Comment by Ash258** [comment]
What is the point of requesting review when you did not do anything about the major issue??

image

**Comment by MunsterPlop** [comment]
My apologies but I'm missing something here.

* Is the problem that it's a nightly build?
If it's that, versions on the ugs repo are not labelled in any way.

* If it's not, can you please clarify what you mean and I will fix it.

**Comment by Ash258** [comment]
It is unacceptable that user is exposed to use nightly build without prior notice. When they do not provide stable builds it will not be added.

**Comment by MunsterPlop** [comment]
Fair enough.
Currently, the link to the latest version and the link to the nightly provide the exact same archive.
I'm gonna check if it's the same problem for their older versions and if it is, I will just close this PR.

**Comment by MunsterPlop** [comment]
Well, it is. Closing.

**Pull request closed without merging**